openmc/surface.py

```diff
@@ -832,36 +778,79 @@ def z0(self, z0):
         check_type('z0 coefficient', z0, Real)
         self._coefficients['z0'] = z0
 
+    def _get_base_coeffs(self):
+        return (0., 0., 1., self.z0)
+
     def bounding_box(self, side):
-        """Determine an axis-aligned bounding box.
+        if side == '-':
+            return (np.array([-np.inf, -np.inf, -np.inf]),
+                    np.array([np.inf, np.inf, self.z0]))
+        elif side == '+':
+            return (np.array([-np.inf, -np.inf, self.z0]),
+                    np.array([np.inf, np.inf, np.inf]))
 
-        An axis-aligned bounding box for surface half-spaces is represented by
-        its lower-left and upper-right coordinates. For the z-plane surface, the
-        half-spaces are unbounded in their x- and y- directions. To represent
-        infinity, numpy.inf is used.
+    def evaluate(self, point):
+        return point[2] - self.z0
+
+
+Plane.register(ZPlane)
+
+
+class QuadricMixin(metaclass=ABCMeta):
+    """A Mixin class implementing common functionality for quadric surfaces"""
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+    @abstractmethod
+    def _get_base_coeffs(self):
+        """Return coefficients a, b, c, d, e, f, g, h, j, k representing a
+        general quadric surface  of the form:
+        :math:`ax^2 + by^2 + cz^2 + dxy + eyz + fxz+ gx + hy + jz +k = 0`.
+
+        """
+        pass
+
+    def get_Abc(self, coeffs=None):
+        """Compute matrix, vector, and scalar coefficients for this surface or
+        for a specified set of coefficients.
 
         Parameters
         ----------
-        side : {'+', '-'}
-            Indicates the negative or positive half-space
+        coeffs : tuple, optional
+            Tuple of coefficients from which to compute the quadric elements.
+            If none are supplied the coefficients of this surface will be used.
+        """
+        if coeffs is None:
+            a, b, c, d, e, f, g, h, j, k = self._get_base_coeffs()
+        else:
+            a, b, c, d, e, f, g, h, j, k = coeffs
+
+        A = np.asarray([[a, d/2, f/2], [d/2, b, e/2], [f/2, e/2, c]])
```

Comment:
If you know the object you're passing is _not_ an array, you might as well go directly to `np.array`
```suggestion
        A = np.array([[a, d/2, f/2], [d/2, b, e/2], [f/2, e/2, c]])
```